api/v1beta1/spannerautoscaler_webhook_test.go

```diff
@@ -40,54 +40,159 @@ var _ = Describe("SpannerAutoscaler validation", func() {
 					ProjectID:  "test-project-id",
 					InstanceID: "test-instance-id",
 				},
-				Authentication: Authentication{
-					IAMKeySecret: &IAMKeySecret{
-						Name:      "test-service-account-secret",
-						Namespace: "",
-						Key:       "secret",
-					},
-				},
-				ScaleConfig: ScaleConfig{
+			},
+		}
+	})
+
+	Describe("Check default settings", func() {
+
+		Describe("Authentication", func() {
+			BeforeEach(func() {
+				testResource.Spec.ScaleConfig = ScaleConfig{
 					ProcessingUnits: ScaleConfigPUs{
 						Min: 1000,
 						Max: 10000,
 					},
-					ScaledownStepSize: 2000,
 					TargetCPUUtilization: TargetCPUUtilization{
 						HighPriority: 30,
 					},
-				},
-			},
-		}
-	})
+				}
+			})
 
-	Context("Check default settings", func() {
-		It("should set authentication type 'gcp-sa-key' automatically", func() {
-			result, err := createResource(testResource)
-			Expect(err).ToNot(HaveOccurred())
-			Expect(result.Spec.Authentication.Type).To(Equal(AuthTypeSA))
-		})
+			Context("IAM key secret is set", func() {
+				BeforeEach(func() {
+					testResource.Spec.Authentication = Authentication{
+						IAMKeySecret: &IAMKeySecret{
+							Name: "test-service-account-secret",
+							Key:  "secret",
+						},
+					}
+				})
 
-		It("should set authentication type 'impersonation' automatically", func() {
-			testResource.Spec.Authentication.IAMKeySecret = nil // unset the value set above
-			testResource.Spec.Authentication.ImpersonateConfig = &ImpersonateConfig{
-				TargetServiceAccount: "dummy@example.com",
-			}
+				Context("namespace exists", func() {
+					BeforeEach(func() {
+						testResource.Spec.Authentication.IAMKeySecret.Namespace = "default"
+					})
 
-			result, err := createResource(testResource)
-			Expect(err).ToNot(HaveOccurred())
-			Expect(result.Spec.Authentication.Type).To(Equal(AuthTypeImpersonation))
-		})
+					It("should set authentication type 'gcp-sa-key' automatically", func() {
+						result, err := createResource(testResource)
+						Expect(err).ToNot(HaveOccurred())
+						Expect(result.Spec.Authentication.Type).To(Equal(AuthTypeSA))
+						Expect(result.Spec.Authentication.IAMKeySecret.Namespace).To(Equal("default"))
+					})
+				})
```

Comment:
Nit: A `BeforeEach` (setup node) is not really needed here. Just moving the `testResource.Spec.Authentication.IAMKeySecret.Namespace = "default"` from `BeforeEach` to `It` (subject node) should be sufficient because it is not being reused any where.